vllm/v1/core/scheduler.py

```diff
@@ -421,13 +425,14 @@ def update_from_output(
 
             cached_encoder_input_ids = (
                 self.encoder_cache_manager.get_cached_input_ids(request))
-            for input_id in list(cached_encoder_input_ids):
-                start_pos = request.mm_positions[input_id]["offset"]
-                num_tokens = request.mm_positions[input_id]["length"]
-                if start_pos + num_tokens <= request.num_computed_tokens:
-                    # The encoder output is already processed and stored
-                    # in the decoder's KV cache.
-                    self.encoder_cache_manager.free(request, input_id)
+            if cached_encoder_input_ids:
```

Comment:
nit: otherwise you actually don't have to change the return type. "if set()" is also False.

```suggestion
            if cached_encoder_input_ids is not None:
```

Reply:
Nice catch! My original intention was to avoid creating many empty sets but I've just found that actually `set()` is almost free. The only thing we need to avoid here is applying `list()` on the empty set. Therefore, I reverted the return type.

Reply:
Maybe a stupid question but why do we need the `list()` at all? Can just iterate over the set?